notebooks/236-raft-optimization/236-raft-optimization.ipynb

```diff
@@ -0,0 +1,570 @@
+    "Convert model to OpenVINO IR."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OpenVINO IR already converted\n"
+     ]
+    }
+   ],
+   "source": [
+    "import subprocess\n",
+    "\n",
+    "model_ov_path = model_onnx_path.with_suffix(\".xml\")\n",
+    "\n",
+    "if not model_ov_path.exists():\n",
+    "    # Run Model Optimizer to convert ONNX model to OpenVINO IR.\n",
+    "    mo_command = f'mo --framework onnx -m {model_onnx_path} --output_dir {base_model_dir}'\n",
+    "    subprocess.call(mo_command, shell=True) \n",
+    "else: print('OpenVINO IR already converted')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "input precision: <Type: 'float32'>\n",
+      "input shape: [1,3,480,640]\n",
+      "output precision: <Type: 'float32'>\n",
+      "output shape: [1,2,480,640]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from openvino.runtime import Core\n",
+    "\n",
+    "core = Core()\n",
+    "core.set_property({'CACHE_DIR': '.cache'})\n",
+    "\n",
+    "model_ov = core.read_model(model=model_ov_path)\n",
+    "compiled_model = core.compile_model(model=model_ov, device_name=\"CPU\")\n",
+    "input_layer = compiled_model.input(0)\n",
+    "output_layer = compiled_model.output(1)\n",
+    "\n",
+    "print(f\"input precision: {input_layer.element_type}\")\n",
+    "print(f\"input shape: {input_layer.shape}\")\n",
+    "N, C, H, W = input_layer.shape\n",
+    "print(f\"output precision: {output_layer.element_type}\")\n",
+    "print(f\"output shape: {output_layer.shape}\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Download and prepare two KITTI frames."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cv2\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from PIL import Image\n",
+    "\n",
+    "def prepare_input(img):\n",
+    "    # plt.imshow(img)\n",
+    "    # plt.show()\n",
+    "    img_input = cv2.resize(img, (W, H))\n",
+    "    img_input = img_input.transpose(2, 0, 1)\n",
+    "    img_input = img_input[np.newaxis,:,:,:]      \n",
+    "\n",
+    "    return img_input.astype(np.float32)\n",
+    "\n",
+    "\n",
+    "IMAGE_PATH_1 = \"../data/image/kitti_1.png\"\n",
+    "IMAGE_PATH_2 = \"../data/image/kitti_2.png\"\n",
+    "\n",
+    "if not Path(IMAGE_PATH_1).exists():\n",
+    "    !wget https://raw.githubusercontent.com/liruoteng/OpticalFlowToolkit/master/data/example/KITTI/frame1.png -O ../data/image/kitti_1.png\n",
```

Comment:
this command will not work on windows, please use `download_file` from utils instead